extensions/amp-dailymotion/1.0/component.js

```diff
@@ -0,0 +1,112 @@
+import * as Preact from '#preact';
+import {forwardRef} from '#preact/compat';
+import {useMemo} from '#preact';
+import {dispatchCustomEvent} from '#core/dom';
+import {
+  DailymotionEvents,
+  getDailymotionIframeSrc,
+  makeDailymotionMessage,
+} from '../dailymotion-api';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {parseQueryString} from '#core/types/string/url';
+
+/**
+ * @param {string} method
+ * @return {string}
+ */
+function makeMethodMessage(method) {
+  switch (method) {
+    case 'mute':
+      return makeDailymotionMessage('muted', [true]);
+    case 'unmute':
+      return makeDailymotionMessage('muted', [false]);
+    case 'showControls':
+      return makeDailymotionMessage('controls', [true]);
+    case 'hideControls':
+      return makeDailymotionMessage('controls', [false]);
+    default:
+      return makeDailymotionMessage(method);
+  }
+}
+
+/**
+ * @param {MessageEvent} event
+ */
+function onMessage({currentTarget, data}) {
+  const parsedData = parseQueryString(/** @type {string} */ (data));
+  const event = parsedData?.['event'];
+  if (event === DailymotionEvents.PAUSE) {
+    dispatchCustomEvent(currentTarget, 'pause');
+  } else if (event === DailymotionEvents.PLAY) {
+    dispatchCustomEvent(currentTarget, 'playing');
+  } else if (event === DailymotionEvents.END) {
+    dispatchCustomEvent(currentTarget, 'pause');
+    dispatchCustomEvent(currentTarget, 'end');
+  } else if (event === DailymotionEvents.API_READY) {
+    dispatchCustomEvent(currentTarget, 'canplay');
+  }
+}
+
+/**
+ * @param {!BentoDailymotionDef.Props} props
+ * @param {{current: ?VideoWrapperDef.Api}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function BentoDailymotionWithRef(
+  {
+    autoplay,
+    endscreenEnable,
+    info,
+    mute,
+    sharingEnable,
+    start,
+    uiHighlight,
+    uiLog,
```

Comment:
```suggestion
    uiLogoDisable,
```

Reply:
As discussed offline, in the interest of parity with the dailymotion documentation and the 0.1 functionality we are leaving these attrs as they are.

Reply:
As we discussed, let's keep these as Dailymotion defines them, where they're true by default. 

Should still be renamed
```suggestion
    uiLogo,
```